File: C/Grace.c
#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#include <stdlib.h>

/*
 Comment here
*/

#define TOPRINT(str) (printf(str, 10, 34, 92, str))
#define REDIRECT(fd) (dup2(fd, STDOUT_FILENO))
#define FT_MAIN() int main(void) { const char *str = "#include <stdio.h>%1$c#include <unistd.h>%1$c#include <fcntl.h>%1$c#include <stdlib.h>%1$c%1$c/*%1$c Comment here%1$c*/%1$c%1$c#define TOPRINT(str) (printf(str, 10, 34, 92, str))%1$c#define REDIRECT(fd) (dup2(fd, STDOUT_FILENO))%1$c#define FT_MAIN() int main(void) { const char *str = %2$c%4$s%2$c; int fd = open(%2$cGrace_kid.c%2$c, O_RDWR | O_CREAT, 0644); if (fd == -1) { write(2, %2$cFailed trying to create/modify Grace_kid.c%3$cn%2$c, 43); exit(1); } if (REDIRECT(fd) == -1) { write(2, %2$cError trying to write the information of the program into Grace_kid.c file%3$cn%2$c, 75); close(fd); exit(1); } close(fd); TOPRINT(str); return (0); }%1$c%1$cFT_MAIN()%1$c"; int fd = open("Grace_kid.c", O_RDWR | O_CREAT, 0644); if (fd == -1) { write(2, "Failed trying to create/modify Grace_kid.c\n", 43); exit(1); } if (REDIRECT(fd) == -1) { write(2, "Error trying to write the information of the program into Grace_kid.c file\n", 75); close(fd); exit(1); } close(fd); TOPRINT(str); return (0); }

FT_MAIN()

File: rust/Grace.rs
use std::fs::File;
use std::io::Write;
use std::process::ExitCode;
/*
 comment in file
*/
macro_rules! print_in_file{($file_object:expr, $string:expr) => { { if let Err(_e) = $file_object.write($string.as_bytes()) { eprintln!("{}", "Writing to Grace_kid.rs failed"); return ExitCode::FAILURE; } } }; }
macro_rules! create_file{() => { { match File::create("Grace_kid.rs") { Ok(v) => v, Err(_e) => { eprintln!("{}", "Failed to create/modify Grace_kid.rs"); return ExitCode::FAILURE; } } } }; }
macro_rules! ft_main{() => { fn main() -> ExitCode { let vector: Vec<char> = vec![10 as char, 34 as char, 123 as char, 125 as char]; let string: String = format!("use std::fs::File;{0}use std::io::Write;{0}use std::process::ExitCode;{0}/*{0} comment in file{0}*/{0}macro_rules! print_in_file{2}($file_object:expr, $string:expr) => {2} {2} if let Err(_e) = $file_object.write($string.as_bytes()) {2} eprintln!({1}{2}{3}{1}, {1}Writing to Grace_kid.rs failed{1}); return ExitCode::FAILURE; {3} {3} {3}; {3}{0}macro_rules! create_file{2}() => {2} {2} match File::create({1}Grace_kid.rs{1}) {2} Ok(v) => v, Err(_e) => {2} eprintln!({1}{2}{3}{1}, {1}Failed to create/modify Grace_kid.rs{1}); return ExitCode::FAILURE; {3} {3} {3} {3}; {3}{0}macro_rules! ft_main{2}() => {2} fn main() -> ExitCode {2} let vector: Vec<char> = vec![10 as char, 34 as char, 123 as char, 125 as char]; let string: String = format!({1}{4}{1}, vector[0], vector[1], vector[2], vector[3], {1}{4}{1}); let file_object = create_file!(); print_in_file!(&file_object, &string); ExitCode::SUCCESS {3} {3}; {3}{0}{0}ft_main!();{0}", vector[0], vector[1], vector[2], vector[3], "use std::fs::File;{0}use std::io::Write;{0}use std::process::ExitCode;{0}/*{0} comment in file{0}*/{0}macro_rules! print_in_file{2}($file_object:expr, $string:expr) => {2} {2} if let Err(_e) = $file_object.write($string.as_bytes()) {2} eprintln!({1}{2}{3}{1}, {1}Writing to Grace_kid.rs failed{1}); return ExitCode::FAILURE; {3} {3} {3}; {3}{0}macro_rules! create_file{2}() => {2} {2} match File::create({1}Grace_kid.rs{1}) {2} Ok(v) => v, Err(_e) => {2} eprintln!({1}{2}{3}{1}, {1}Failed to create/modify Grace_kid.rs{1}); return ExitCode::FAILURE; {3} {3} {3} {3}; {3}{0}macro_rules! ft_main{2}() => {2} fn main() -> ExitCode {2} let vector: Vec<char> = vec![10 as char, 34 as char, 123 as char, 125 as char]; let string: String = format!({1}{4}{1}, vector[0], vector[1], vector[2], vector[3], {1}{4}{1}); let file_object = create_file!(); print_in_file!(&file_object, &string); ExitCode::SUCCESS {3} {3}; {3}{0}{0}ft_main!();{0}"); let file_object = create_file!(); print_in_file!(&file_object, &string); ExitCode::SUCCESS } }; }

ft_main!();
